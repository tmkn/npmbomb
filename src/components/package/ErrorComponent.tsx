/** @jsx jsx */
import { jsx, css } from "@emotion/core";
import React, { useRef, useState, useContext } from "react";
import { useHistory } from "react-router-dom";

import { AppContext } from "../../App";
import { mq, textColor } from "../../css";
import { Center } from "../shared/center/Center";
import { PrimaryButton } from "../shared/buttons/Buttons";

export const ErrorComponent: React.FC<{ pkgName: string }> = ({ pkgName }) => {
    const { setAppState } = useContext(AppContext);
    const history = useHistory();

    const style = css({
        [mq[0]]: {
            color: textColor,
            margin: "1rem 0",
            flex: 1,
            backgroundColor: "#FBE9E7",
            padding: "1rem"
        }
    });

    function onClick(): void {
        setAppState({
            gameMode: false,
            guesses: [],
            remaining: []
        });
        history.push("/");
    }

    return (
        <React.Fragment>
            <h1>{pkgName}</h1>
            <Center>
                <div css={style}>
                    <span>Whoops, couldn't load data!</span>
                </div>
            </Center>
            <Center>
                <PrimaryButton onClick={onClick}>Home</PrimaryButton>
            </Center>
        </React.Fragment>
    );
};
