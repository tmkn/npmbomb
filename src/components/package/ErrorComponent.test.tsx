import React from "react";
import { BrowserRouter } from "react-router-dom";
import { render, fireEvent, waitForElement } from "@testing-library/react";

import { ErrorBanner, ErrorComponent, NotFound } from "./ErrorComponent";

describe("<ErrorComponent />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(
            <BrowserRouter>
                <ErrorComponent pkgName={"test"} />
            </BrowserRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});

describe("<NotFound />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(
            <BrowserRouter>
                <NotFound pkgName={"test"} />
            </BrowserRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});

describe("<ErrorBanner />", () => {
    test("matches snapshot", () => {
        const { asFragment } = render(
            <BrowserRouter>
                <ErrorBanner>hello</ErrorBanner>
            </BrowserRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
