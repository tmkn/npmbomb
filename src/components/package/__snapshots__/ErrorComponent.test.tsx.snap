// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBanner /> matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-8a5j6a"
  >
    <div
      class="css-1x66upb"
    >
      hello
    </div>
  </div>
</DocumentFragment>
`;

exports[`<ErrorComponent /> matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    test
  </h1>
  <div
    class="css-8a5j6a"
  >
    <div
      class="css-1x66upb"
    />
  </div>
  <div
    class="css-8a5j6a"
  >
    <div
      class="css-16wgzhr"
    >
      Home
    </div>
  </div>
</DocumentFragment>
`;

exports[`<NotFound /> matches snapshot 1`] = `
<DocumentFragment>
  <h1>
    test
  </h1>
  <div
    class="css-8a5j6a"
  >
    <div
      class="css-1x66upb"
    >
      <span>
        Whoops, couldn't find data for this package!
      </span>
      <br />
      <span>
        <span
          class="css-9wkshj"
        >
          Propose this package
        </span>
      </span>
    </div>
  </div>
  <div
    class="css-8a5j6a"
  >
    <div
      class="css-16wgzhr"
    >
      Home
    </div>
  </div>
</DocumentFragment>
`;
